src/controllers/hawkerCentreController.ts

```diff
@@ -3,7 +3,9 @@ import {Request, Response, NextFunction} from 'express';
 
 async function retrieveHawkerCentre(req: Request, res: Response, next: NextFunction) {
   try {
-    const hawkerCentre = await HawkerCentre.findByPk(req.params.id);
+    const hawkerCentre = await HawkerCentre.findByPk(req.params.id, {
```

Comment:
Abstract the attributes option out as it is duplicated in `retrieve` and `index`

Reply:
A better way is to create another middleware that transform the `id` to `hawkerCentreId` and attach it at the end of call stack in `routes`.

Reply:
Pull the original data out of the database and just change the `id`, then don't have to explicitly state what attributes you have to return. Why -> imagine if we add another new attribute to `hawkerCentre`, then need change here also

Reply:
And also the `index` function